test/tests/bcs/ad_nodal_bc_nonlocal_dependence/test.i

```diff
@@ -0,0 +1,60 @@
+[Mesh]
+  [gen]
+    type = GeneratedMeshGenerator
+    dim = 1
+    nx = 20
+  []
+  [pin]
+    type = ExtraNodesetGenerator
+    input = gen
+    new_boundary = 'pin'
+    nodes = '0'
+  []
+[]
+
+[Variables]
+  [u][]
+[]
+
+[Kernels]
+  [diff]
+    type = Diffusion
+    variable = u
+  []
+[]
+
+[BCs]
+  [left_neumann]
+    type = NeumannBC
+    boundary = 'left'
+    value = -1
+    variable = u
+  []
+  [right_neumann]
+    type = NeumannBC
+    boundary = 'right'
+    value = 1
+    variable = u
+  []
+  [pin]
+    type = ADAverageValuePin
+    variable = u
+    boundary = 'pin'
```

Comment:
```suggestion
    type = ADAverageValuePin
    variable = u
    # This BC actually constrains all the domain, not just the boundary
    boundary = 'pin'
```

Reply:
It really does only constrain the single degree of freedom on the boundary `pin` in order to remove this problem's nullspace. But by constraining the single dof it does ensure that the average value in the entire domain is 0
```suggestion
    type = ADAverageValuePin
    variable = u
    # This BC constrains a single degree of freedom on the node specified by the boundary name 'pin' but since 
    # only gradients of 'u' appear in the weak form, constraining the single degree of freedom will make it appear as 
    # if we have constrained all the degrees of freedom in the system such that we have an average value of 0. 
     boundary = 'pin'
```